### PR TITLE
feat(BILL-CPC-1317): displayed customer current balance redo

### PR DESCRIPTION
**Pull request by MercuryAtom31**
JIRA: link to jira ticket

https://champlainsaintlambert.atlassian.net/browse/CPC-1317

## Context:
What is the ticket about and why are we doing this change.

We want to allow customers to see their current balance on the billing main page.

## Does this PR change the .vscode folder in petclinic-frontend?:
If the PR changes the .vscode folder, explain why in detail because it should not. 
Be sure to include cgerard321 as a reviewer.

It does not.

<span style="color:red">**Reviewers need to check for any changes to the 
.vscode folder and add a comment about it to their review comments.**</span>

## Changes
What are the various changes and what other modules do those changes affect.

Files Modified

This can be bullet point or sentence format.
## Before and After UI (Required for UI-impacting PRs)
If this is a change to the UI, include before and after screenshots to show the differences.
If this is a new UI feature, include screenshots to show reviewers what it looks like. 

Screenshot 2024-10-17 at 21 19 10

## Dev notes (Optional)
Specific technical changes that should be noted
## Linked pull requests (Optional)
Pull request links

### CODE REVIEW

**Review by ctsaki25 — COMMENTED** [review]
why are you changing the package-lock.json and package.json?

**Comment by MercuryAtom31** [comment]
> why are you changing the package-lock.json and package.json?

Because of the react-counterup's installation.

**Review by Futurespast — APPROVED** [review]

File: api-gateway/src/test/java/com/petclinic/bffapigateway/presentationlayer/v2/CustomerBillControllerIntegrationTest.java [hunk 1]
```diff
@@ -70,11 +73,50 @@ public void stopMockServer() {
...
+    // String invalidCustomerId = "invalid-id";
+
+    // webTestClient.get()
+    // .uri("/api/v2/gateway/customers/{customerId}/bills/current-balance",
+    // invalidCustomerId)
+    // .cookie("Bearer", MockServerConfigAuthService.jwtTokenForValidOwnerId)
+    // .accept(MediaType.APPLICATION_JSON)
+    // .exchange()
+    // .expectStatus().isNotFound();
+    // }
```
Comment (ThomasBedard) [inline api-gateway/src/test/java/com/petclinic/bffapigateway/presentationlayer/v2/CustomerBillControllerIntegrationTest.java:121]: Everything looks great, however, I am not sure it's good practice to put commented out testing code to main.

**Review by ThomasBedard — DISMISSED** [review]

File: billing-service/src/test/java/com/petclinic/billing/businesslayer/BillServiceImplTest.java [hunk 2]
```diff
@@ -612,6 +612,41 @@ void getBillsByMonth_ShouldReturnBillsForGivenMonth() {
                 .verifyComplete();
     }
 
+    @Test
```
Comment (vinivelozo) [inline billing-service/src/test/java/com/petclinic/billing/businesslayer/BillServiceImplTest.java:615]: Great tests

**Review by vinivelozo — DISMISSED** [review]
Great job, Looks good and did not touch the vscode. Just be sure to remove the code that you don't use ( the ones that is comment out ).

**Pull request merged**